rllib/algorithms/algorithm.py

```diff
@@ -576,65 +592,12 @@ def setup(self, config: PartialAlgorithmConfigDict):
 
         # Evaluation WorkerSet setup.
         # User would like to setup a separate evaluation worker set.
-
-        # Update with evaluation settings:
-        user_eval_config = copy.deepcopy(self.config["evaluation_config"])
```

Comment:
Evaluation config is now an automatically generated (full) AlgorithmConfig inside the main AlgorithmConfig (property: `self.evaluation_config`). The `self.evaluation_config` property within the evaluation config is None. This helps us do all validation checking early on (before freezing) and simplifies handling of eval configs.

The typical eval-config override (via a dict) is still supported, but will eventually be replaced by objects (e.g. a to-be-designed `AlgorithmConfigOverride` class) as well.